src/integration/java/io/pinecone/helpers/TestIndexResourcesManager.java

```diff
@@ -168,7 +204,12 @@ private String createOrGetCollection() throws InterruptedException {
     private void seedIndex(List<String> vectorIds, String indexName) throws InterruptedException {
         // Build upsert request
         Index indexClient = pineconeClient.getIndexConnection(indexName);
-        indexClient.upsert(buildRequiredUpsertRequestByDimension(vectorIds, dimension), "");
+
+        // Upsert vectors into both custom and default namespaces
+        List<String> namespaces = Arrays.asList(customNamespace, defaultNamespace);
+        for (String namespace : namespaces) {
+            indexClient.upsert(buildRequiredUpsertRequestByDimension(vectorIds, dimension), namespace);
+        }
 
         // Wait for record freshness
```

Comment:
Now that we're upserting multiple, do we want to check for the total number of vectors we expect before continuing?

Reply:
I'm actually about to change this ,but I'll move this logic to `createOrGetServerlessIndex` instead